config/formats.ts

```diff
@@ -1370,12 +1374,14 @@ export const Formats: FormatList = [
 			};
 			const tier = this.toID(species.tier) || 'ou';
 			if (!(tier in boosts)) return;
-			const pokemon: Species = this.dex.deepClone(species);
+			const pokemon = this.dex.deepClone(species);
+			pokemon.bst = 0;
 			const boost = boosts[tier];
 			let statName: StatName;
-			for (statName in pokemon.baseStats) {
+			for (statName in pokemon.baseStats as StatsTable) {
```

Comment:
why is this necessary?

Reply:
Because `pokemon` is no longer a `Species` (so that I can change its `bst`), `pokemon.baseStats` is no longer a `StatsTable`, so `StatName` is no longer the key of it, resulting in the error "The left-hand side of a 'for...in' statement must be of type 'string' or 'any'".